hamocc/mo_inventory_bgc.F90

```diff
@@ -0,0 +1,883 @@
+       &  +zocetratot(izoo)+zocetratot(iphosph)                                &
+       &  +zpowtrato(ipowaph)+zsedlayto(issso12)                               &
+       &  +zburial(issso12)                                                    &
+       &  +zprorca
+
+  totalsil=                                                                    &
+       &   zocetratot(isilica)+zocetratot(iopal)                               &
+       &  +zpowtrato(ipowasi)+zsedlayto(issssil)+zburial(issssil)              &
+       &  +zsilpro
+
+  totaloxy=                                                                    &
+       &  (zocetratot(idet)+zocetratot(idoc)+zocetratot(iphy)                  &
+       &  +zocetratot(izoo))*(-24.)+zocetratot(ioxygen)                        &
+       &  +zocetratot(iphosph)*2 +zocetratot(isco212)+zocetratot(icalc)        &
+       &  +zocetratot(iano3)*1.5+zocetratot(ian2o)*0.5                         &
+       &  +zsedlayto(issso12)*(-24.) + zsedlayto(isssc12)                      &
+!       &  +zburial(issso12)*(-24.)   +   zburial(isssc12)                      &
+       &  +zpowtrato(ipowno3)*1.5+zpowtrato(ipowaic)                           &
+       &  +zpowtrato(ipowaox)+zpowtrato(ipowaph)*2                             &
+       &  +zprorca*(-24.)+zprcaca                                              &
+#if defined(BOXATM)
+       &  +zatmo2*ppm2con+zatmco2*ppm2con
+#else
+       & +so2flux+sn2oflux*0.5+co2flux
+#endif
+
+! Compute sediment fluxes
+!----------------------------------------------------------------------
+  sum_zprorca  = 0.
+  sum_zprcaca  = 0.
+  sum_zsilpro  = 0.
+  sum_sedfluxo = 0.
+  sum_expoor   = 0.
+  sum_expoca   = 0.
+  sum_exposi   = 0.
+
+  call sum2d(kpie,kpje,dlxp,dlyp,bgct2d(:,:,jprorca),sum_zprorca, mask=weto)
+  call sum2d(kpie,kpje,dlxp,dlyp,bgct2d(:,:,jprcaca),sum_zprcaca, mask=weto)
+  call sum2d(kpie,kpje,dlxp,dlyp,bgct2d(:,:,jsilpro),sum_zsilpro, mask=weto)
+
+  call sum2d_array(kpie,kpje,npowtra,dlxp,dlyp,sedfluxo,sum_sedfluxo)
+
+  call sum2d(kpie,kpje,dlxp,dlyp,expoor,sum_expoor)
+  call sum2d(kpie,kpje,dlxp,dlyp,expoca,sum_expoca)
+  call sum2d(kpie,kpje,dlxp,dlyp,exposi,sum_exposi)
+
+! Write output to netCDF file or stdout
+!----------------------------------------------------------------------
+  if (mnproc == 1) then
+     if (GLB_INVENTORY(iogrp) == 2) then
+        call write_netcdf(iogrp)
+     else
+        call write_stdout
+     end if
+  end if
+
+END SUBROUTINE INVENTORY_BGC
+
+
+subroutine sum2d(kpie,kpje,dlxp,dlyp,var2d,total,mask)
+!**********************************************************************
+!**** Sum 2D scalar fields
+!**********************************************************************
+  implicit none
+  integer, intent(in) :: kpie, kpje
+  real, dimension(kpie,kpje), intent(in) :: dlxp, dlyp, var2d
+  real, intent(out) :: total
+  real, dimension(kpie,kpje), intent(in), optional :: mask
```

Comment:
Do we have a case where it makes sense to call this routine without "mask"? If not this could be simplified by making mask non-optional

Reply:
mask is now required in sum2d routine.